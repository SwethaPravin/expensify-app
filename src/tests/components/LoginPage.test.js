import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { LoginPage } from '../../components/LoginPage';

test ('should render Login Page',() => {
  const wrapper = shallow(<LoginPage startLogin={ ()=>{} }/>);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

test('should call startLogin when login button clicked', () => {
  const startLogin = jest.fn();
  const wrapper = shallow(<LoginPage startLogin={ startLogin }/>);
  wrapper.find('button').simulate('click');
  expect(startLogin).toHaveBeenCalled();
});