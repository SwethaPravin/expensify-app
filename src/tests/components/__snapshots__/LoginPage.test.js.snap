// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login Page 1`] = `
<div>
  <h1>
    Expensify Login Page
  </h1>
  <br />
  <button
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
